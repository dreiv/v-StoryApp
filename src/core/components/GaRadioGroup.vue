<script setup lang="ts">
import { computed, provide, useCssModule } from 'vue'
import { uniqueId } from '../utils/uniqueId'
import { radioGroupKey } from '../constants'

export interface RadioGroupProps {
  title?: string
  name?: string
  direction?: 'horizontal' | 'vertical'
}

const { direction = 'vertical', name = uniqueId('group') } = defineProps<RadioGroupProps>()
const style = useCssModule()

const classes = computed(() => [style.content, { [style.horizontal]: direction === 'horizontal' }])

provide(radioGroupKey, { name })
</script>

<template>
  <fieldset :class="$style.group">
    <legend :class="$style.title">
      <slot name="title">{{ title }}</slot>
    </legend>

    <div :class="classes">
      <slot />
    </div>
  </fieldset>
</template>

<style module>
.group {
  margin: 0;
  border: none;
  padding: 0;

  .title {
    margin-bottom: var(--ga-size-8);
    padding-left: 0;

    font-weight: var(--font-weight-medium);
    line-height: var(--text-md--line-height);
    font-family: var(--default-font-family);
  }

  .content {
    display: flex;
    flex-direction: column;
    gap: var(--ga-size-8);

    &.horizontal {
      flex-direction: row;
    }
  }
}
</style>
