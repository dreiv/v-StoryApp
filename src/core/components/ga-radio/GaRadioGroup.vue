<script setup lang="ts">
import { computed, provide, useCssModule } from 'vue'
import { radioGroupKey } from '@/core/constants'
import { uniqueId } from '@/core/utils/uniqueId'

export interface RadioGroupProps {
  title?: string
  name?: string
  direction?: 'horizontal' | 'vertical'
}

const { direction = 'vertical', name = uniqueId('group') } = defineProps<RadioGroupProps>()
const model = defineModel<string>({ default: '' })
const style = useCssModule()

const classes = computed(() => [style.content, { [style.horizontal]: direction === 'horizontal' }])

provide(radioGroupKey, { name, model })
</script>

<template>
  <fieldset :class="$style.group">
    <legend :class="$style.title">
      <slot name="title">{{ title }}</slot>
    </legend>

    <div :class="classes">
      <slot />
    </div>
  </fieldset>
</template>

<style module>
.group {
  margin: 0;
  border: none;
  padding: 0;
}

.title {
  padding-left: 0;

  font-weight: var(--font-weight-medium);
  line-height: var(--text-md--line-height);
  font-family: var(--default-font-family);

  &:not(:empty) {
    margin-bottom: var(--ga-size-spacing-03);
  }
}

.content {
  display: flex;
  flex-direction: column;
  gap: var(--ga-size-spacing-03);

  &.horizontal {
    flex-direction: row;
  }
}
</style>
