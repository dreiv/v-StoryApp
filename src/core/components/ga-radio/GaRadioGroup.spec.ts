import { shallowMount, type VueWrapper } from '@vue/test-utils'
import GaRadioGroup from './GaRadioGroup.vue'

vi.mock('./types', () => ({ radioGroupKey: 'radioGroup' }))

describe('RadioGroup', () => {
  let wrapper: VueWrapper

  beforeEach(() => {
    wrapper = shallowMount(GaRadioGroup)
  })

  it('should render correctly with default props', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with title when provided', async () => {
    await wrapper.setProps({ title: 'Test Title' })
    expect(wrapper.text()).toContain('Test Title')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with title slot when provided', () => {
    wrapper = shallowMount(GaRadioGroup, {
      slots: { title: '<div class="custom-title">Custom Title</div>' },
    })
    expect(wrapper.find('.custom-title').exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with horizontal direction when specified', async () => {
    await wrapper.setProps({ direction: 'horizontal' })
    expect(wrapper.find('.horizontal').exists()).toBe(true)
  })
})
