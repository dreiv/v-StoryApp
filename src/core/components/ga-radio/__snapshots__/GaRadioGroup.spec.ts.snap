// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RadioGroup > should render correctly with default props 1`] = `
"<fieldset class="group">
  <legend class="title"></legend>
  <div class="content"></div>
</fieldset>"
`;

exports[`RadioGroup > should render with title slot when provided 1`] = `
"<fieldset class="group">
  <legend class="title">
    <div class="custom-title">Custom Title</div>
  </legend>
  <div class="content"></div>
</fieldset>"
`;

exports[`RadioGroup > should render with title when provided 1`] = `
"<fieldset class="group">
  <legend class="title">Test Title</legend>
  <div class="content"></div>
</fieldset>"
`;
